src/coffea/ml_tools/torch_wrapper.py

```diff
@@ -0,0 +1,81 @@
+import warnings
+
+import numpy
+
+try:
+    import torch
+except ImportError as err:
+    warnings.warn(
+        "Users should make sure the torch package is installed before proceeding!\n"
+        "> pip install torch\n"
+        "or\n"
+        "> conda install torch",
+        UserWarning,
+    )
+    raise err
+
+from .helper import lazy_container, numpy_call_wrapper
+
+
+class torch_wrapper(lazy_container, numpy_call_wrapper):
+    """
+    Wrapper for running pytorch with awkward/dask-awkward inputs.
+    """
+
+    def __init__(self, torch_jit: str):
+        """
+        As torch models are guaranteed to be directly pickle-able depending on
+        model, we load the model using torch file save states. Notice that we
+        only support TorchScript files for this wrapper class [1]. If the user
+        is attempting to run on the clusters, the TorchScript file will need to
+        be passed to the worker nodes in a way which preserves the file path.
+
+        [1]
+        https://pytorch.org/tutorials/beginner/saving_loading_models.html#export-load-model-in-torchscript-format
+
+        Parameters
+        ----------
+
+        - torch_jet: The TorchScript file to load for inference
+        """
+        lazy_container.__init__(self, ["model", "device"])
+
+        # Saving the file path to be loaded by dask workers.
+        self.torch_jit = torch_jit
+
+    def _create_device(self):
+        """
+        Torch device run calculations on. This wrapper class will always attempt
+        to use GPU if possible. Setting this as a lazy object so that remote
+        worker can have a different configuration the interactive session.
+        """
+        # Setting up the default torch inference computation device
+        return torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    def _create_model(self):
+        """
+        Creating the model instance of the using either the weakref or the
+        dask.persist instance (prioritizes using weakref). Enforcing the eval
+        mode after initialization.
+        """
+        if torch.cuda.is_available():
+            model = torch.jit.load(self.torch_jit).cuda()
+        else:
+            model = torch.jit.load(self.torch_jit)
+        model.eval()
+        return model
```

Comment:
The way this is implemented currently it will deserialize the entire model for each chunk (when N_chunks is large). Significantly inefficiency. Would prefer it rolls out without scaling issues but we can also fix it in another PR.

Ditto for xgboost. Triton is fine for understood reasons. :-)

Reply:
However, I think this one can also come later.